### PR TITLE
[idea][bug 477183] Find Usages for property syntax

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=477183

**Comment by akosyakov** [comment]
Jenkins intellij-tests-master-latest

File: plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.eclipse.xtext.common.types.JvmFeature
+import org.eclipse.xtext.common.types.JvmOperation
+
+import static extension java.beans.Introspector.*
+import static extension java.lang.Character.*
+
+/**
+ * @author kosyakov - Initial contribution and API
+ */
+class PropertyUtil {
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend]: Please use the same logic as in org.eclipse.xtext.xbase.scoping.batch.AbstractSessionBasedScope.toProperty(String, JvmFeature, int, int)

Comment (szarnekow) [inline plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend]: My, bad, looks fine :)

File: plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	def static String getPropertyName(JvmFeature feature) {
+		if (feature.static) {
+			feature.getPropertyName(feature.simpleName, 1, 2)
+		} else {
+			feature.getPropertyName(feature.simpleName, 0, 1)
+		}
+	}
+
+	def static String getPropertyName(JvmFeature feature, String methodName, int getterParams, int setterParams) {
+		if (feature instanceof JvmOperation)
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend]: This looks unnecessarily complex to me, e.g each invocation will create an array of pairs to filter et al. This is called quite often thus shouldn't do any fancy tricks.

**Pull request merged**